typing/out_type.ml

```diff
@@ -650,16 +654,60 @@ let tree_of_type_path p =
    we maintain the invariant that the row variable uniquely identifies the type.
    This, however, means, we can no longer distinguish between the row variable
    and the type containing the row variable. *)
-let proxy ty = Transient_expr.repr (proxy ty)
+module Proxy : sig
+  type t
 
-(* When printing a type scheme, we print weak names.  When printing a plain
-   type, we do not.  This type controls that behavior *)
-type type_or_scheme = Type | Type_scheme
+  val make : type_expr -> t
+
+  val desc : t -> type_desc
+
+  val is_non_gen : type_or_scheme -> t -> bool
 
-let is_non_gen mode ty =
-  match mode with
-  | Type_scheme -> is_Tvar ty && get_level ty <> generic_level
-  | Type        -> false
+  val id : t -> int
+
+  val refresh : t -> t
+
+  module Set : Stdlib.Set.S with type elt = t
+  module Map : Stdlib.Map.S with type key = t
+
+end = struct
+  module T = struct
+    type t = transient_expr
+
+    let compare = TransientTypeOps.compare
+  end
+  include T
+
+  module Set = Stdlib.Set.Make(T)
+  module Map = Stdlib.Map.Make(T)
```

Comment:
Aren't those not `TransientTypeSet` and `TransientTypeMap` ?

Reply:
Yes, will fix